### PR TITLE
GitHub actions

### PR DESCRIPTION
**Pull request by ylavoie**
This PR enables CI testing on Github Actions, including:
- testing on Chrome, Opera, Firefox and PhantomJS
- using `yath` to run each test
Coverage is computed but not published, because CircleCI handles publishing on coveralls

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -49,76 +119,150 @@ jobs:
...
+        uses: actions/cache@v2
+        with:
+          # A list of files, directories, and wildcard patterns to cache and restore
+          path: /opt/hostedtoolcache/perl
+          # An explicit key for restoring and saving the cache
+          key: cpan-${{ matrix.perl }}-${{ hashFiles('cpanfile') }}
+
+      - name: Install required features
+        run: |
+          cpm install -g -v --resolver=metacpan --notest \
```
Comment (ehuelsmann) [inline .github/workflows/main.yml:133]: I found `cpm` to be much faster than `cpanm` however, the latter allows doesn't install dists which have a satisfied version already installed. `cpm` simply installs all packages named in the list of dependencies, no exceptions.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -49,76 +119,150 @@ jobs:
...
         run: |
-          sed -e "s#{ROOT}#$PWD#" doc/conf/webserver/nginx-travis.conf >/tmp/nginx.conf
-          nginx -c /tmp/nginx.conf &
+          starman --preload-app --pid starman.pid --workers $JOB_COUNT \
+                  --max-requests 5000 --error-log logs/starman-error.log \
+                  -Ilib -Iold/lib --port 5762 bin/ledgersmb-server.psgi &
 
-      - name: Starting 'PhantomJS'
-        run: phantomjs --webdriver=4422 2>/dev/null >/dev/null &
+      # Fix the condition to debug
```
Comment (ehuelsmann) [inline .github/workflows/main.yml:219]: Yes :-)

File: UI/lib/ui-header.html [hunk 3]
```diff
@@ -16,26 +16,25 @@
...
-    [%+# HTML Snippet, for import only %]
-    [%+#
+    [%-#
         # source comment only!
         #
         # don't specify a title on the stylesheets: we want them to be
         # *persistent*
         # http://www.w3.org/TR/html401/present/styles.html#h-14.3.1
-    %]
+    -%]
```
Comment (ehuelsmann) [inline UI/lib/ui-header.html:25]: Is there a reason you removed all the minus signs? I mean, they are there to strip the leading spaces, in order to result in a "nice" HTML output.

Comment (ylavoie) [reply UI/lib/ui-header.html:25]: I added them for that exact purpose.

File: webpack.config.js [hunk 4]
```diff
@@ -235,10 +242,10 @@ const lsmbCSS = {
 const VirtualModulePluginOptions = {
     moduleName: "js-src/lsmb/bootstrap.js",
     contents:
-       `/* eslint-disable */
-        define(["dojo/parser","dojo/ready","`
-        + includedRequires.join('","')
-        + `"], function(parser, ready) {
+        `/* eslint-disable */
+        define(["dojo/parser","dojo/ready","` +
+        includedRequires.join('","') +
```
[the remark this answers is not in the document]
Comment (ylavoie) [reply webpack.config.js]: Why indeed? Fixed.

File: UI/lib/ui-header.html (hunk 3 above)
Comment (ehuelsmann) [reply UI/lib/ui-header.html:25]: Oops...

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -49,76 +119,150 @@ jobs:
...
+          LANG: en_US.UTF-8
+        if: ${{ matrix.BROWSER == 'phantomjs' }}
 
-      - name: Starting 'starman'
-        run: starman --preload-app --pid starman.pid --workers 2 -Ilib -Iold/lib --port 5762 bin/ledgersmb-server.psgi &
+      - name: Create default test database
+        run: |
+          echo "connect_data:"            > $HOME/.ledgersmb-admin.conf
+          echo "  user: $PGUSER"         >> $HOME/.ledgersmb-admin.conf
+          echo "  password: $PGPASSWORD" >> $HOME/.ledgersmb-admin.conf
```
[the remark this answers is not in the document]
Comment (ylavoie) [reply .github/workflows/main.yml]: FIxed

File: .circleci/config.yml [hunk 6]
```diff
@@ -65,6 +69,33 @@ commands:
...
+            echo '"Time","CPU","Memory"' > $FILE;
+            #read -rst5 j memory_limit < <(grep hierarchical_memory_limit /sys/fs/cgroup/memory/memory.stat);
+            #CircleCI free plan gives 2 processors & 4GB of RAM
+            memory_limit=4294967296
+            n_procs=2
+            read -rst5 cpu_usage0     </sys/fs/cgroup/cpu/cpuacct.usage;
+            read -rst5 cstart         < <(date +"%s%N");
+            while true; do
+              sleep 1;
+              read -rst5 memory_usage </sys/fs/cgroup/memory/memory.usage_in_bytes;
```
[the remark this answers is not in the document]
Comment (ylavoie) [reply .circleci/config.yml]: Done

**Pull request merged**